index.js

```diff
@@ -103,11 +103,11 @@ function copySharedLibs (builds, folder, opts, cb) {
   })
 }
 
-function run (cmd, cwd, env, cb) {
+function run (cmd, opts, cb) {
   if (!cmd) return cb()
 
-  var shell = os.platform() === 'android' ? 'sh' : undefined
-  var child = execspawn(cmd, {cwd: cwd, env: env, stdio: 'inherit', shell: shell})
+  var shell = opts.platform === 'android' ? 'sh' : undefined
```

Comment:
@ralphtheninja In the case of the `leveldown` cross-compile setup, where does this code run? Inside the dockcross Docker container, right? Which is not Android, but Debian? I'm a bit confused 😄

Reply:
I don't know the story behind this line of code but to answer your question:

`prebuildify-cross` runs https://github.com/ahdinosaur/prebuildify-cross/blob/master/build-in-docker#L16 inside the docker container defined by https://github.com/ahdinosaur/prebuildify-cross/blob/master/Dockerfile, which in turn is based on `dockercross/${TARGET}` (for example https://github.com/dockcross/dockcross/blob/master/linux-armv7/Dockerfile.in in the `linux-armv7` case) which in turn is based on `dockercross/base` which in turn is based on `debian/jessie` https://github.com/dockcross/dockcross/blob/master/Dockerfile.in#L1 , so yes, it's based on Debian :smile:

Reply:
So I guess https://github.com/ahdinosaur/prebuildify-cross/blob/master/build#L241 needs to change and instead do `--env PREBUILD_PLATFORM=${PLATFORM}` (after this has been merged).

Reply:
> I don't know the story behind this line of code 

I think there are two use cases:

1. Create prebuilds in Android (in which case you need `sh`)
2. Create prebuilds through cross-compiling (in our case, running in Debian, and it should use the default shell here?)

So we need a distinction between "target platform" (for which `PREBUILD_PLATFORM` is intended) and "platform this code is running on" (i.e. simply `os.platform()`).

Reply:
> Hm, I guess I don't understand why we need to override platform at all..

Yeah that's a really good point. For our purposes of cross compiling `leveldown` it will always be `linux` right?